app/models/facility_group.rb

```diff
@@ -53,14 +55,43 @@ def toggle_diabetes_management
     end
   end
 
-  def report_on_patients
-    Patient.where(registration_facility: facilities)
-  end
-
   def diabetes_enabled?
     facilities.where(enable_diabetes_management: false).count.zero?
   end
 
+  def create_state_region!
+    return unless Flipper.enabled?(:regions_prep)
+    return if state_region || state.blank?
+
+    Region.state_regions.create!(name: state, reparent_to: organization.region)
+  end
+
+  def update_block_regions!
+    create_block_regions!
+    remove_block_regions!
+  end
+
+  def create_block_regions!
+    return unless Flipper.enabled?(:regions_prep)
+    return if new_blocks.blank?
+
+    new_blocks.map { |name|
+      Region.block_regions.create!(name: name, reparent_to: region)
+    }
```

Comment:
Idea: we return a `.all?` on the `.map` so we can chain the methods in the controller. That will get rid of the `if .. true else false`.